ambry-rest/src/main/java/com.github.ambry.rest/NettySslFactory.java

```diff
@@ -166,19 +166,17 @@ private static TrustManagerFactory getTrustManagerFactory(SSLConfig config)
    * @return the list of supported cipher suites, or {@code null} if the configs did not specify any
    */
   private static Iterable<String> getCipherSuites(SSLConfig config) {
-    ArrayList<String> cipherSuitesList = Utils.splitString(config.sslCipherSuites, ",");
-    return (cipherSuitesList.size() > 0 && !(cipherSuitesList.size() == 1 && cipherSuitesList.get(0).isEmpty()))
-        ? cipherSuitesList : null;
+    List<String> cipherSuitesList = Utils.splitString(config.sslCipherSuites, ",");
+    return cipherSuitesList.size() > 0 ? cipherSuitesList : null;
   }
 
   /**
    * @param config the {@link SSLConfig}.
    * @return the list of supported cipher suites, or {@code null} if the configs did not specify any
    */
   private static String[] getEnabledProtocols(SSLConfig config) {
-    String[] enabledProtocols = config.sslEnabledProtocols.split(",");
-    return enabledProtocols.length > 0 && !(enabledProtocols.length == 1 && enabledProtocols[0].isEmpty())
-        ? enabledProtocols : null;
+    List<String> enabledProtocols = Utils.splitString(config.sslEnabledProtocols, ",");
+    return !enabledProtocols.isEmpty() ? enabledProtocols.toArray(new String[0]) : null;
```

Comment:
original code returns `enabledProtocols`, why here changed to `enabledProtocols.toArray(new String[0])`?

Reply:
I was using `Array::split` before which meant that the content was already in an array. I switched to using the `Utils::splitString` convenience method which requires converting back to an array.